### ISSUE TITLE
dxi validation rules in DxForm not triggered when a custom template is used.

### ISSUE THREAD

**Issue by briosheje** [DevExpress/devextreme-angular#802]
**To help us process the issue more efficiently, please provide the following information:**
- Specify the version of the *devextreme-angular* and *devextreme* packages you are using.

    devexteme version: ^18.1.3
    devextreme-angular version: ^18.1.3
- Specify the type of the issue (check one with "x"):
[x] bug Report 

I can't reproduce on the plunkr because some dependencies seems to be unavailable, but here is the issue I'm encountering:

Assuming a generic form like that:
```
<dx-form
      [(formData)]="myFormData"
    >
      <dxi-item 
          dataField="name"
          label="Name"
          >
          <dxi-validation-rule
              type="required"
              message="name is mandatory">
          </dxi-validation-rule>
      </dxi-item>
      <dxi-item 
          dataField="myCustomField"
          label="Custom Field"
          template="myCustomTemplate"
          >
          <dxi-validation-rule
              type="required"
              message="name is mandatory">
          </dxi-validation-rule>
      </dxi-item>
    <div *dxTemplate="let data of 'myCustomTemplate'">
         <input [(ngModel)]="myFormData['myCustomField']" />
    </div>
    </dx-form>
    
    
    <dx-button [text]="buttonText"></dx-button>
```

having some random data in it, the validation rules seems to not trigger for the item with a custom template.

More precisely, if you happen to have any item rendered with a custom template that handles a form field element, the validation rule never triggers when the value is changed (despite it is, in fact, changed in the form model).

In a nutshell, the data binding works properly, but the validation does not trigger, despite I would expect it to work because of the type "required" no validation rules, which should just check whether the model value is defined, right?

If that's an intended behavior, is there any other way to manipulate form errors so that using DxForm along with custom widgets can be feasible?

Another (perhaps related) noticable side-effect is that whenever you use a custom template, the `onFieldDataChanged` event of the form widget is **not** triggered if a form value is changed within the custom template.

**Comment by DokaRus** [comment]
Hi,

>If that's an intended behavior, is there any other way to manipulate form errors so that using DxForm along with custom widgets can be feasible?

Whenever you defined an item template you're overwriting the default Form validation logic. We have discussed it in our Support Center, please inspect the [T518854 ticket](https://www.devexpress.com/Support/Center/Question/Details/T518854/dxform-validation-for-item-with-template-angular). 

>Another (perhaps related) noticable side-effect is that whenever you use a custom template, the onFieldDataChanged event of the form widget is not triggered if a form value is changed within the custom template.

The form doesn't know how the template's content operates with its data and when it's necessary to fire an `onFieldDataChanged` event. I can propose you use the API `updateData` method, like in [this example](https://next.plnkr.co/edit/y1sfNdcpvo0xIe41?preview).

**Comment by briosheje** [comment]
@DokaRus  thanks for the fast feedback.

Looking from the sample, I already do add the validations manually, but I don't link them to the same validation group. I will try to put them all in the same validation group and validate them manually, which is not (of course) an issue.

Perhaps it would be worth to point that out in the documentation in this section: https://js.devexpress.com/Documentation/ApiReference/UI_Widgets/dxForm/Item_Types/SimpleItem/#template

Thanks for the support.

**Comment by DokaRus** [comment]
You're welcome.
Thank you too for your feedback. I'll discuss it with the team.
